packages/completer/src/handler.ts

```diff
@@ -111,17 +111,18 @@ class CompletionHandler implements IDisposable {
    * Get whether the completion handler is disposed.
    */
   get isDisposed(): boolean {
-    return this._completer === null;
+    return this._isDisposed;
   }
 
   /**
    * Dispose of the resources used by the handler.
    */
   dispose(): void {
-    this._completer = null;
-
-    // Use public accessor to disconnect from editor signals.
-    this.editor = null;
+    if (this.isDisposed) {
```

Comment:
There's no problem with this check per se, but `Signal.clearData(...)` is a no-op when called multiple times, so the check is not strictly necessary.